c++/cmake/CapnProtoConfig.cmake.in

```diff
@@ -1,19 +1,47 @@
+# Cap'n Proto CMake Package Configuration
+#
+# When configured and installed, this file enables client projects to find Cap'n Proto using
+# CMake's find_package() command. It adds imported targets in the CapnProto:: namespace, such as
+# CapnProto::kj, CapnProto::capnp, etc. (one target for each file in pkgconfig/*.pc.in), defines
+# the capnp_generate_cpp() function, and exposes some variables for compatibility with the original
+# FindCapnProto.cmake module.
+#
 # Example usage:
 #   find_package(CapnProto)
 #   capnp_generate_cpp(CAPNP_SRCS CAPNP_HDRS schema.capnp)
 #   include_directories(${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
I know this isn't your change, but I think this should be:

    include_directories(${CAPNP_HDRS})

Reply:
I didn't realize you could pass full file paths to `include_directories()`! Will verify and fix, as that would indeed be better.

Reply:
Your surprise makes me think I have `CAPNP_HDRS` confused with `CAPNPC_OUTPUT_DIR`!

Reply:
I think `${CMAKE_CURRENT_BINARY_DIR}` is correct but you should change:
`include_directories(${CMAKE_CURRENT_BINARY_DIR})`
to
`target_include_directories(YOUR_TARGET_NAME PRIVATE ${CMAKE_CURRENT_BINARY_DIR})`
or by
`set(CMAKE_INCLUDE_CURRENT_DIR ON)`
The latter add current source directory and current binary dir.

Reply:
@pqu that makes sense. If `CAPNPC_OUTPUT_DIR` is guaranteed to be set for the user (unless overridden, of course), then that would be more precise than using `CMAKE_CURRENT_BINARY_DIR`. Also, I agree with @damienrg about preferring `target_include_directories()`. I'll research and update when I can.

Ultimately, I'd like to ~steal~ copy protobuf's `protobuf_generate(TARGET)` function, which would allow us to work in terms of targets, rather than source and header files.

Reply:
`CAPNPC_OUTPUT_DIR` is [always set](https://github.com/capnproto/capnproto/blob/97376af85649739acc5278ef39d64dac3d3381b0/c%2B%2B/cmake/CapnProtoMacros.cmake#L82) by `CAPNP_GENERATE_CPP` if not already.

Reply:
Unfortunately, `capnp_generate_cpp()` doesn't use `set(... PARENT_SCOPE)`, so we can't rely on `CAPNPC_OUTPUT_DIR` always being set. Do you think we should change it to use `PARENT_SCOPE`?

My preference would be to avoid the issue by adopting protobuf's cleaner target-based function, as then there'd be no need for any `include_directories()` call at all.

Reply:
I agree that a target-based function would be preferable. Are there backwards-compatibility concerns with re-writing the interface to `CAPNP_GENERATE_CPP` – or should it be deprecated for a new `CAPNP_GENERATE` (?) function that uses a target-based interface?

Reply:
We could probably make it backwards-compatible, so that existing calls will continue to work, but a new target-based mode could be added. Augmenting the current interface to the function might have ramifications on the implementation of the function, for better or worse -- I guess we'll see.

I'll have to study protobuf's solution here. Their `protobuf_generate()` function might support languages other than C++, which might be the real reason why they chose to add a new function.

I'll open a separate issue for this.